### PR TITLE
refactor: Migration of Loan Application screen to compose

### LINKED ISSUE #2546: refactor: Migration of Loan Application Screen from XML to Compose
**Summary:**

 Migration of Loan Application Screen from XML to Compose

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
Fixes #2546 
Jira Task: [MM-14](https://mifosforge.jira.com/browse/MM-14)

Please Add Screenshots If there are any UI changes.
| XML Screen (before)                                                                                               |  Compose Screen (Now)                                                                                            | 
|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| https://github.com/openMF/mifos-mobile/assets/77102514/0c1d054d-2048-462a-9a7e-d4786b3dc90e | https://github.com/openMF/mifos-mobile/assets/77102514/c4c9cbc1-a182-4de6-88a4-1419d31576ee |

[MM-14]: https://mifosforge.jira.com/browse/MM-14?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+            labelResId = R.string.purpose_of_loan,
+            onClick = { index, item ->
+                selectPurpose(index)
+                selectedLoanPurpose = item
+            }
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        OutlinedTextField(
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt]: MifosOutlinedTextField

File: app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            onDrawableClick = { showDatePicker = true }
+        )
+
+        Spacer(modifier = Modifier.height(20.dp))
+
+        Button(
+            onClick = {
+                selectedLoanProductError = null
+                principalAmountError = null
+                if(uiData.selectedLoanProduct.isNullOrBlank()) {
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt]: replace with when

File: app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+    if (showDatePicker) {
+        DatePickerDialog(
+            modifier = Modifier.padding(horizontal = 20.dp),
+            onDismissRequest = { showDatePicker = false },
+            confirmButton = {
+                TextButton(
+                    onClick = {
+                        val timeInMillis = datePickerState.selectedDateMillis
+                        val formattedDate = SimpleDateFormat("dd-MM-yyyy", Locale.getDefault()).format(timeInMillis)
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt]: move to utility function

File: app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                        expectedDisbursementDate = formattedDate
+                        setDisbursementDate(formattedDate)
+                        showDatePicker = false
+                    }
+                ) { Text("OK") }
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt]: string res

File: app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+            },
+            dismissButton = {
+                TextButton(
+                    onClick = { showDatePicker = false }
+                ) { Text("Cancel") }
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationContent.kt]: ...

File: app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationFragment.kt [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        viewModel.loadLoanTemplate()
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MifosMobileTheme {
+                    LoanApplicationScreen(
+                        viewModel = viewModel,
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationFragment.kt]: hilt viewmodel in  compose

File: app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationScreen.kt [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        NoInternet(
+            icon = R.drawable.ic_portable_wifi_off_black_24dp,
+            error = R.string.no_internet_connection,
+            isRetryEnabled = true,
+            retry = onRetry
+        )
+    } else {
+        Toast.makeText(context, stringResource(id = R.string.error_fetching_template), Toast.LENGTH_SHORT).show()
+    }
+}
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/loan_account_application/LoanApplicationScreen.kt]: make a generic composable

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

**Comment by Aditya-gupta99** [comment]
LGTM

**Pull request merged**